firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -107,12 +107,21 @@
   }
 
   /**
-   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and {@link
-   * FirebaseOptions#getProjectId()} are not null.
+   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and ({@link
+   * FirebaseOptions#getProjectId()} or {@link FirebaseOptions#getGcmSenderId()} ()} are not null.
    */
   private void preConditionChecks() {
+    Preconditions.checkNotNull(getApplicationId());
+    Preconditions.checkNotNull(getProjectIdentifier());
```

Comment:
How about checkNotEmpty?
Reason: If the value is set to "" it doesn't help us either.

Reply:
Done.